File: python/fibonacci.py
def calcula_fibonacci(numero):
    fibonacci = [0, 1]
    while fibonacci[-1] < numero:
        fibonacci.append(fibonacci[-1] + fibonacci[-2])
    return fibonacci


def solicitar_numero():
    while True:
        try:
            numero = int(input("Digite um número: "))
            return numero
        except ValueError:
            print("Por favor, insira um número inteiro válido.")


numero = solicitar_numero()
fibonacci = calcula_fibonacci(numero)

if numero in fibonacci:
    print(f"Sequência de Fibonacci até {numero}: {fibonacci}")
    print(f"{numero} pertence à sequência de Fibonacci.")
else:
    print(f"{numero} não pertence à sequência de Fibonacci.")

File: nodejs/fibonacci.js
function calculaFibonacci(numero) {
  const fibonacci = [0, 1];

  while (fibonacci[fibonacci.length - 1] < numero) {
    const [a, b] = fibonacci.slice(-2);
    fibonacci.push(a + b);
  }

  return fibonacci;
}

function solicitarNumero() {
  const prompt = require("prompt-sync")();

  while (true) {
    const input = prompt("Digite um número: ");
    const numero = parseInt(input);

    if (!isNaN(numero)) {
      return numero;
    }

    console.log("Por favor, insira um número inteiro válido.");
  }
}

const numero = solicitarNumero();
const fibonacci = calculaFibonacci(numero);

if (fibonacci.includes(numero)) {
  console.log(`Sequência de Fibonacci até ${numero}: [${fibonacci}]`);
  console.log(`${numero} pertence à sequência de Fibonacci.`);
} else {
  console.log(`${numero} não pertence à sequência de Fibonacci.`);
}
